### PR TITLE
Usage Charge "ID on Null" Fix

### PR DESCRIPTION
**Pull request by Kyon147**
Currently, the usage Charge method is trying to get the user from the session but it is not available.

This updates the route to expect the shop domain to be passed in like the other functions.

### CODE REVIEW

**Comment by Kyon147** [comment]
Hey @osiset 

I've updated the usage charge wiki as it was not passing over the signature as a string but as it is breaking `Call to a member function getId() on null` after some digging, the user is never in the session. 

As the other methods, just pass in the shop domain, it made sense to just keep it consistent. 

When merged I'll update the wiki to pass in the shop domain also. 

Issues relate: 

#1166
#1044 
#1002

**Review by osiset — COMMENTED** [review]
One comment there.. up to you if you think its a better method or not.

**Comment by nahid** [comment]
Is there any need to add `verify.shopify` middleware to the `usage_charge` route when we pass the `shop` param with the URL?

File: src/Traits/BillingController.php [hunk 1]
```diff
@@ -100,11 +102,24 @@ public function process(
...
      * @return RedirectResponse
      */
-    public function usageCharge(StoreUsageCharge $request, ActivateUsageCharge $activateUsageCharge): RedirectResponse
-    {
+    public function usageCharge(
+        StoreUsageCharge    $request,
+        ActivateUsageCharge $activateUsageCharge,
+        ShopQuery           $shopQuery
+    ): RedirectResponse {
+        // Get the shop from the shop param after it has been validated.
```
[the remark this answers is not in the document]
Comment (Kyon147) [reply src/Traits/BillingController.php:116]: Good thinking on this, updated this now 👍

**Comment by Kyon147** [comment]
@nahid no need to pass the verify middleware as that is why the shop param is passed along.

**Review by osiset — APPROVED** [review]
Good by me.

**Comment by Kyon147** [comment]
@osiset merged in #1222 now so if you are happy, will merge this into master and do a small patch release

**Pull request merged**